source/extensions/filters/common/expr/context.h

```diff
@@ -106,12 +106,65 @@ class WrapperFieldValues {
 
 using WrapperFields = ConstSingleton<WrapperFieldValues>;
 
+// Forward declarations
 class RequestWrapper;
+class ResponseWrapper;
+class ConnectionWrapper;
+class UpstreamWrapper;
+class PeerWrapper;
+class FilterStateWrapper;
+class XDSWrapper;
+
+// Type aliases for value extractors
+using CelValueExtractor = std::function<absl::optional<CelValue>(const RequestWrapper&)>;
+using ResponseValueExtractor = std::function<absl::optional<CelValue>(const ResponseWrapper&)>;
+using ConnectionValueExtractor = std::function<absl::optional<CelValue>(const ConnectionWrapper&)>;
+using UpstreamValueExtractor = std::function<absl::optional<CelValue>(const UpstreamWrapper&)>;
+using XDSValueExtractor = std::function<absl::optional<CelValue>(const XDSWrapper&)>;
+using SslExtractor = std::function<CelValue(const Ssl::ConnectionInfo&)>;
+
+// Forward declare the singleton value classes
+class RequestLookupValues {
+public:
+  const absl::flat_hash_map<absl::string_view, CelValueExtractor> request_lookup_;
+  static const RequestLookupValues& get();
+};
+
+class ResponseLookupValues {
+public:
+  const absl::flat_hash_map<absl::string_view, ResponseValueExtractor> response_lookup_;
+  static const ResponseLookupValues& get();
+};
+
+class ConnectionLookupValues {
+public:
+  const absl::flat_hash_map<absl::string_view, ConnectionValueExtractor> connection_lookup_;
```

Comment:
string_view doesn't own the data and could trigger lifetime issue like used in container case. I guess here is fine as long as you ensure the strings that are pointed to by string_view outlast the string_view itself.

But I am just curious any specific gain here by using string_view?

Reply:
Memory savings? The key is not stored as a copy. I think this is safe since these are all constexpr keys.

Reply:
@tyxia @kyessenov Is it okay to keep this as `string_view`?